### ISSUE TITLE
FluxC: Make sure to completely erase all info on logout

### ISSUE THREAD

**Issue by daniloercoli** [wordpress-mobile/WordPress-Android#5304]
- Logged out from the app, and closed it for real (removed it from the background apps list).
- Restarted the app from scratch.
```
E/Volley: [1233] BasicNetwork.performRequest: Unexpected response code 403 for https://public-api.wordpress.com/rest/v1.1/sites/XXXX?locale=it_IT
I/WordPress-NOTIFS: Sending GCM token to our remote services: XXXXXXXXXX
E/Volley: [1233] BasicNetwork.performRequest: Unexpected response code 403 for https://public-api.wordpress.com/rest/v1.1/sites/XXXXXX?locale=it_IT
E/WordPress-API: Volley error
                 com.android.volley.AuthFailureError
                     at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:171)
                     at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:114)
D/WordPress-API: Dispatching action: SiteAction-UPDATE_SITE
I/WordPress-NUX: onSiteChanged: org.wordpress.android.fluxc.store.SiteStore$OnSiteChanged@14a95bfc
E/WordPress-API: onSiteChanged has error: GENERIC_ERROR - org.wordpress.android.fluxc.store.SiteStore$SiteError@70b2285
```

I see a call to the .com REST API, (+ the GCM token is sent to our server), even if i'm not logged in the app.

**Comment by aforcier** [comment]
@daniloercoli do you know if this was a clean install of the app, or if you had migrated from an earlier version to FluxC?

**Comment by aforcier** [comment]
Also I think the GCM token part predates FluxC: I see it even in a clean install of `alpha-36` (pre-FluxC), on first run of the app (not logged in to anything).

**Comment by daniloercoli** [comment]
Don't remember exactly, probably it was a migration. Anyway, I can't replicate it on a clean installation now.

> the GCM token part predates FluxC

Ok, lt close this ticket then, and open a new one for the token thing.

**Comment by aforcier** [comment]
As discussed, re-opening it for the WP.com logout part of it, as it might be related to some migration token issues we're seeing.

**Comment by aforcier** [comment]
Could not reproduce after an `alpha-36` to `alpha-37` migration.

**Comment by maxme** [comment]
Probably related to https://github.com/wordpress-mobile/WordPress-Android/issues/5315

**Comment by oguzkocer** [comment]
Since we closed #5315 due to the fixed id bug in FluxC, should we close this too?

**Comment by aforcier** [comment]
I think we can @oguzkocer, @daniloercoli already opened a separate issue for the GCM token part: https://github.com/wordpress-mobile/WordPress-Android/issues/5311.
